x-pack/plugins/rule_registry/server/event_log/log/event_query.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { estypes } from '@elastic/elasticsearch';
+import { set } from '@elastic/safer-lodash-set';
+import { Logger } from 'kibana/server';
+import { IIndexReader } from '../elasticsearch';
+import { truthy } from '../utils/predicates';
+import { IEventQuery } from './public_api';
+
+export interface EventQueryParams {
+  indexReader: IIndexReader;
+  logger: Logger;
+  baseRequest: estypes.SearchRequest;
+}
+
+export class EventQuery<TEvent> implements IEventQuery<TEvent> {
+  private readonly indexReader: IIndexReader;
+  private readonly baseRequest: estypes.SearchRequest;
+
+  constructor(params: EventQueryParams) {
+    this.indexReader = params.indexReader;
+    this.baseRequest = params.baseRequest;
+  }
+
+  public async execute(): Promise<TEvent[]> {
+    const response = await this.indexReader.search<TEvent>(this.baseRequest);
+    return response.body.hits.hits.map((hit) => hit._source).filter(truthy);
+  }
+
+  public async search<TDocument = TEvent>(
+    extraRequest: estypes.SearchRequest
+  ): Promise<estypes.SearchResponse<TDocument>> {
+    const finalRequest = mergeRequests(this.baseRequest, extraRequest);
```

Comment:
I agree, separating these two APIs makes sense to avoid confusion that might be caused by the hidden logic of merging requests. Also, this logic would not exist => less code, fewer potential bugs. Great suggestion 👍

Reply:
Addressed